From 5cc6f0e73d8bbdcced36f19df4b5fb38b96cc788 Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Tue, 17 Sep 2019 18:00:54 +0200
Subject: [PATCH 058/103] mm: slab: BUG on page type confusion under
 BUG_ON_DATA_CORRUPTION

This change was extracted from PaX.

Signed-off-by: Daniel Micay <danielmicay@gmail.com>
Signed-off-by: Levente Polyak <levente@leventepolyak.net>
Signed-off-by: Thibaut Sautereau <thibaut.sautereau@ssi.gouv.fr>
---
 mm/slab.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/mm/slab.h b/mm/slab.h
index 95eb34174..dc6ab73aa 100644
--- a/mm/slab.h
+++ b/mm/slab.h
@@ -639,9 +639,13 @@ static inline struct kmem_cache *virt_to_cache(const void *obj)
 	struct slab *slab;
 
 	slab = virt_to_slab(obj);
+#ifdef CONFIG_BUG_ON_DATA_CORRUPTION
+	BUG_ON(!slab);
+#else
 	if (WARN_ONCE(!slab, "%s: Object is not a Slab page!\n",
 					__func__))
 		return NULL;
+#endif
 	return slab->slab_cache;
 }
 
-- 
2.36.1.203.g1bcf4f6271

